cve_bin_tool/checkers/perl.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+
+"""
+CVE checker for Perl
+References:
+https://www.cvedetails.com/vulnerability-list/vendor_id-1885/Perl.html
+"""
+from ..util import regex_find
+
+
+def get_version(lines, filename):
+    """
+    returns version information for perl as found in a given file.
+    
+    VPkg: perl, perl
+    """
+    regex = [
+        r"perl-[45]:([5]+\.[0-9]+\.[0-9]+)",
+        r"perl-[45]:([5]+\.[0-9]+)",
+        r"perl [45]+, version [0-9]+, subversion [0-9]+",
```

Comment:
Getting closer, but regex_find isn't going to know what to do with this last regex to get a single version, because it's expecting everything in the round brackets () to be something that looks pretty close to a version.  You didn't include any round brackets, and even if you did it's going to have a bunch of other stuff in there that doesn't make a clean version string of the type we're expecting.   I don't think it's worth changing regex_find for this, so probably you'll want to have special logic just for perl (as in, keep your changes in this file, don't change util for a single case).  

You may want to read up on python's regex stuff before attempting this https://docs.python.org/3/library/re.html.  In particular, you're likely going to want to use re.sub which will let you substitute a new string.  Let's open up a python console and see how that would work:

```
>>> perlstring = "This is perl 5, version 26, subversion 2"
>>> re.sub(r"This is perl (5), version (26), subversion (2)", r"\1.\2.\3", perlstring)
'5.26.2'
```

The "magic" here is that in that second string, \1 refers to the first thing in brackets, \2 refers to the second and so on, so we can make a string that is just first.second.third and it will do the right thing.  I'm using a regex that matches the string exactly (it has to be perl 5, version 26, etc.) but you've already got a more generalized one up above that you can use (just don't forget to put the brackets in for match groups).  Open up a python console and play around with different perlstrings so you can see how it works.

And then you've got to integrate it into checker. I'd suggest you take that regex out of the regex_find list and search for it separately so you can use `re.sub` directly.  If you look at some of the other checkers you can see that sometimes we do a check to see if our version is still unknown and then refine like this.